standards/component-documentation.md

```diff
@@ -0,0 +1,58 @@
+(see examples below)
+
+### Detailed description
+After the interface declaration, we should provide a more detailed description of the component. This should include how the component works, and how it should be used. This section should be written in a more technical style, and should be more detailed than the short non-technical description. Here you can use the names of the interfaces you declared earlier.
+
+Examples of stuff to discuss are:
+- How to use the component
+- What conventions you used for LSB marking, signed binary number formats, etc.
+- With what delay your component responds to inputs
+- What the limits of the component are, and what happens if you exceed them
+- Things that might cause undefined or buggy behavior
+- Whether or not it is okay to "merge delete" the input and output interfaces to reduce the delay of the component
+- Links to external resources like data sheets or more thorough explanations of the component if they don't fit nicely in a blueprint description
+
+### Contact information, license, and credits
+Finally, the blueprint description should include contact information for the creator of the component (probably you). Preferably, this should include a Discord username, but something like an email or a statement that you want messages via steam is also fine.
+Here you can also give other people permission to use your component, and if and how you want to be credited for it.
+Of course if you used someone else's creation or idea in your component, you should credit them here, even if you got permission from them.
+
+## Example
+```
+This is a 4-bit adder. It takes two 4-bit numbers as input, and outputs the sum of those numbers. This is useful for adding two numbers together in a computer.
+
+blue 4-bit input : A
+red 4-bit input : B
+cyan 1-bit input : Carry in
+green 4-bit output : Sum
+yellow 1-bit output : Carry out
+
+Add two 4-bit two's complement numbers together. The carry in is added to the least significant bit of the first number. The sum is output as a 4-bit two's complement number, and the carry out is the carry out of the most significant bit, acting like a 5th output bit. The carry in is optional, and can be left unconnected if not needed.
+
+This is a carry lookahead adder, and it has a delay of 6 ticks. It is balanced, so streaming in numbers at a high rate should not cause any issues. The input gates are just plain and-gates with no other inputs, so merge-deleting them is fine. The output gates are XOR gates with no looped back outputs, so they can be merge-deleted as well, as long as they are merged with other XOR gates.
+
+This was made by CodeMaker_4, you can contact me via Discord on @codemaker_4. You can use this component in your creations, but please credit me if you do by mentioning my name and the name of this component in the description of your creation. This is my own work, based on the relevant wikipedia article.
+```
```

Comment:
This example should be the a prime example of how to use the standard, therefore we should make it a bit more exact and verbose. While also keeping it somewhat concise as to not discourage people from using this standard.
- Input A and B should be renamed to something like "operand A" and "operand B" and later described.
- It should be stated that "the adder can add unsigned numbers and negative numbers in two's complement". Right now it could be assumed (although this would be weird) that the adder that is described can only add negative numbers in two's complement representation and not positive numbers.